### DISCUSSION TITLE
Is CatBoostModel#predict thread-safe?

### DISCUSSION THREAD

**Discussion by elenaryz** [catboost/catboost#2711] [Q&A]
And does it make sense to run predict in parallel using single model loaded?

**Comment by andrey-khropov** [comment]
What API are you interested in?

Python, R, JVM, Node.js?

**Comment by andrey-khropov** [comment]
Judging by `CatBoostModel` + `predict`, I assume it is JVM applier. Yes, it should be possible. 

See also: 

https://tech.deliveryhero.com/is-catboost-faster-than-lightgbm-and-xgboost/
https://github.com/catboost/catboost/issues/2548#issuecomment-1887414839
